tests/test_morphing.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+
+from madminer.utils import morphing as m
+
+
+
+
+def original_find_components(parameter_max_power = [2,2]):
+    # Test find_components with max_power = [2,2], without max_overall power
+    morpher = m.PhysicsMorpher(parameter_max_power=parameter_max_power)
+    components = morpher.find_components()
+    expected_components = np.array([[0,0],[0,1],[0,2],[1,0],[1,1],[1,2],[2,0],[2,1],[2,2]])
+    assert components.any() == expected_components.any(), "The output of original_find_components is different from expected"
```

Comment:
I think you what you should use here is `np.array_equal`, as `.any()` essentially will be `True` if you have a non-zero value in the array, so this will almost always be satisfied otherwise.

Reply:
This similarly applies to other tests in this file.

Reply:
My bad.
The values between my hand inputs and the actual calculated value have small difference, should i just restrict the comparison to a certain digit after decimal point or calculate the difference and test if the difference is greater than a certain value, 10^-3 for example. 
<img width="382" alt="image" src="https://user-images.githubusercontent.com/67325944/187978552-a711d9e8-c667-4efa-bc94-9e5f601792cf.png">

Reply:
If you see a small difference (and understand the origin), then you can use `np.allclose()` with `atol` or `rtol` arguments to relax the tolerance of the comparison.

Reply:
I think it is better to not manually take the difference, but feed the reference and the calculated array both into `np.allclose`, as that ultimately is more readable as "we compare these two" than having the extra step of taking the difference and comparing that against a threshold.

Reply:
Sure, indeed. 
I changed the code into this
<img width="442" alt="image" src="https://user-images.githubusercontent.com/67325944/187981560-5b771e3a-e7ee-4c94-bfbb-2c8afd9102e3.png">

Reply:
Yes, this looks correct. There are default values for the tolerances (see the documentation https://numpy.org/doc/stable/reference/generated/numpy.allclose.html), but it does not hurt to specify them explicitly.